vpr/src/place/timing/PlacerSetupSlacks.cpp

```diff
@@ -0,0 +1,92 @@
+
+#include "PlacerSetupSlacks.h"
+
+#include "timing_util.h"
+#include "timing_info.h"
+
+
+PlacerSetupSlacks::PlacerSetupSlacks(const ClusteredNetlist& clb_nlist,
+                                     const ClusteredPinAtomPinsLookup& netlist_pin_lookup,
+                                     std::shared_ptr<const SetupTimingInfo> timing_info)
+    : clb_nlist_(clb_nlist)
+    , pin_lookup_(netlist_pin_lookup)
+    , timing_info_(std::move(timing_info))
+    , timing_place_setup_slacks_(make_net_pins_matrix(clb_nlist_, std::numeric_limits<float>::quiet_NaN())) {
+}
+
+/**
+ * @brief Updated the setup slacks in the timing_place_setup_slacks_ data structure.
```

Comment:
Shouldn't this comment be in the .h file?  Suggest moving it there, or deleting it if it is a duplicate.